lib/temple/html/attribute_merger.rb

```diff
@@ -6,40 +6,35 @@ class AttributeMerger < Filter
       define_options merge_attrs: {'id' => '_', 'class' => ' '}
 
       def on_html_attrs(*attrs)
-        names = []
         values = {}
 
-        attrs.each do |attr|
-          name, value = attr[2].to_s, attr[3]
+        attrs.each do |_, _, name, value|
+          name = name.to_s
           if values[name]
             raise(FilterError, "Multiple #{name} attributes specified") unless options[:merge_attrs][name]
             values[name] << value
           else
             values[name] = [value]
-            names << name
           end
         end
 
-        attrs = names.map do |name|
-          value = values[name]
+        attrs = values.map do |name, value|
           if (delimiter = options[:merge_attrs][name]) && value.size > 1
             exp = [:multi]
             if value.all? {|v| contains_nonempty_static?(v) }
               exp << value.first
               value[1..-1].each {|v| exp << [:static, delimiter] << v }
-              [:html, :attr, name, exp]
             else
               captures = unique_name
               exp << [:code, "#{captures} = []"]
               value.each_with_index {|v, i| exp << [:capture, "#{captures}[#{i}]", v] }
               exp << [:dynamic, "#{captures}.reject(&:empty?).join(#{delimiter.inspect})"]
             end
-            [:html, :attr, name, exp]
           else
-            [:html, :attr, name, value.first]
+            exp = value.first
           end
+          [:html, :attr, name, exp]
```

Comment:
not sure if taking above 3 lines and always returning this line is easier to understand or not